### PR TITLE
CI: add st-a5 and st-sim-a5 DFX per-feature smoke steps.

### LINKED ISSUE #1249: [Code Health] Harden a5 DFX scene tests: PMU artifact validation gaps + missing st-sim-a5 smoke coverage
### Category

Robustness (potential edge-case failure)

### Component

Tests

### Description

During review of the commit that mirrors the a2a3 DFX scene tests to a5, four
follow-up hardening points surfaced. None block the mirror (it is a faithful
copy), but each weakens the coverage the DFX tests are supposed to provide.

**A. PMU artifact validation is too lenient** (issues 1–3 below live in
`test_pmu.py::_validate_pmu_artifact`; the a2a3 and a5 files are byte-identical
apart from arch strings, so all three exist in **both** and should be fixed in
both to avoid divergence):

1. **Silent skip on capture failure.** `if not matches: return` — when PMU
   capture fails so badly that no output directory is produced, the glob
   matches nothing and the method returns without any assertion, so the test
   *passes* on a real capture regression instead of failing.

2. **Stale-directory risk.** `matches[-1]` picks the most-recent directory by
   mtime but never confirms it belongs to the current test invocation. A
   directory left over from a prior run/session with the same case label
   satisfies the glob and gets validated in place of (or in addition to) the
   current run's real output — masking a current-run capture regression.

3. **Header ordering not verified.** The docstring claims the smoke asserts the
   header "starts with the documented prefix" (leading, ordered), but the loop
   only checks `col in header_cols` (membership). A header with the same columns
   in a different order passes incorrectly. Either tighten the check to a
   leading ordered-prefix comparison, or correct the docstring to match the
   membership-only intent.

**B. CI coverage gap — `st-sim-a5` runs no DFX per-feature smoke steps.**
`.github/workflows/ci.yml`'s `st-sim-a5` job runs only the default
`pytest examples tests/st --platform a5sim ...` with no `--enable-*` flags.
The a5 DFX cases are no-ops without their flag, so they are **collected but not
validated** in sim CI. By contrast `st-sim-a2a3` mounts four dedicated steps
(`dep_gen`, `l2_swimlane`, `PMU`, `args_dump`). Note the onboard job already
mirrors all four a5 smokes — the gap is **only on the sim side**.

### Location

```markdown
- `tests/st/a5/tensormap_and_ringbuffer/dfx/pmu/test_pmu.py:94-96` (silent skip + stale dir)
- `tests/st/a5/tensormap_and_ringbuffer/dfx/pmu/test_pmu.py:16` (docstring) vs `:102-103` (impl) — header order
- `tests/st/a2a3/tensormap_and_ringbuffer/dfx/pmu/test_pmu.py` — same three, fix in lockstep with a5
- `.github/workflows/ci.yml` — `st-sim-a5` job (~L250-307) missing DFX smoke steps present in `st-sim-a2a3` (~L226-247)
```

### Proposed Fix

1. Replace the `if not matches: return` early-out with an assertion that at
   least one matching output directory exists.
2. Bind the validated directory to the current invocation (e.g. capture a
   pre-run timestamp/marker and require `mtime`/name newer than it, or record
   the exact output path the run created rather than globbing by label).
3. Either assert `header_cols[:len(prefix)] == list(prefix)` for a true ordered
   prefix, or relax the docstring to state membership-only.
4. Add the four DFX smoke steps to `st-sim-a5`, mirroring the `st-sim-a2a3`
   set with `--platform a5sim`, `--clone-protocol https`, and
   `tests/st/a5/...` paths (graphviz for dep_gen is already installed in the
   a5 sim job).

Fix items 1–3 in a single commit touching both a2a3 and a5 so they stay in
sync.

### Priority

Medium (minor risk, should fix in next few releases)

### PR DESCRIPTION
**Pull request by yanghaoran29**
Fixes https://github.com/hw-native-sys/simpler/issues/1249

The a2a3/a2a3sim job already runs four dedicated DFX smoke steps (dep_gen, l2_swimlane, PMU, args_dump) with the matching --enable-* flags. The a5/a5sim job only ran the default pytest sweep, so a5 DFX scene tests were collected but their capture pipelines were never validated in sim CI.

Mirror the st-a2a3/st-sim-a2a3 smoke set in st-a5/st-sim-a5 with a5/a5sim paths and flags. graphviz is already installed in the a5/a5sim job for dep_gen.

Items A.1-A.3 (PMU artifact validation hardening in test_pmu.py) were already landed on main; this commit closes the remaining sim CI gap.

Also update a5's DeviceRunner::finalize_collectors() to use finalize() instead of stop() for diagnostics collectors, matching a2a3's implementation. This ensures shared memory is properly released on any exit from run() — success or early error — preventing stale state issues when workers are reused across runs (e.g., pytest session-scoped worker pool). Add scope_stats_collector_ cleanup with args.scope_stats_data_base reset.

### CODE REVIEW

File: src/a5/platform/onboard/host/device_runner.cpp [hunk 1]
```diff
@@ -748,17 +748,26 @@ int DeviceRunner::finalize() {
...
+    // otherwise re-enter init_l2_swimlane() with stale state still allocated.
+    // Matches a2a3's finalize_collectors().
     if (l2_swimlane_collector_.is_initialized()) {
-        l2_swimlane_collector_.stop();
+        l2_swimlane_collector_.finalize(/*unregister_cb=*/nullptr, prof_free_cb);
     }
     if (dump_collector_.is_initialized()) {
-        dump_collector_.stop();
+        dump_collector_.finalize(/*unregister_cb=*/nullptr, prof_free_cb);
     }
```
Comment (ChaoZheng109) [inline src/a5/platform/onboard/host/device_runner.cpp:745]: onboard `finalize_collectors()` 现在做的是完整 finalize,但 `finalize()`(约 647–662 行)仍保留着一份等价的内联 finalize 序列,两处逻辑重复。

建议统一:把 `finalize()` 里 `:647-662` 这段替换成直接调用 `finalize_collectors()`,与
- a2a3 onboard(`finalize()` → `finalize_collectors()`,`device_runner.cpp:703`)
- 本 PR 的 a5 sim(`:643`)

保持一致,消除重复,避免以后两处漂移。功能上等价(两条路径都会 finalize 释放 shm,且双调用安全),纯属结构去重。

Comment (yanghaoran29) [reply src/a5/platform/onboard/host/device_runner.cpp:745]: 已按建议去重:把 `finalize()` 里 `:647-662` 的内联 finalize 序列替换成直接调用 `finalize_collectors()`,与 a2a3 onboard(`device_runner.cpp:703`)和本 PR 的 a5 sim(`:643`)保持一致。功能等价(is_initialized() 守卫保证 run() 退出路径已 finalize 后的二次调用为 no-op)。已提交于 `ff475e7a`。

**Review by ChaoZheng109 — APPROVED** [review]
